src/network/networkd-link.c

```diff
@@ -2829,6 +2848,23 @@ int link_update(Link *link, sd_netlink_message *m) {
                         return r;
         }
 
+        /* find whether link is enslaved */
+        r = sd_netlink_message_read_u32(m, IFLA_MASTER, &master);
+        if (r >= 0) {
+
+                r = sd_netlink_message_enter_container(m, IFLA_PROTINFO);
+                if (r >= 0) {
+                        uint8_t state;
+
+                        r = sd_netlink_message_read_u8(m, IFLA_BRPORT_STATE, &state);
+                        if (r >= 0)
+                                link->bridge_port = true;
+
+                        (void) sd_netlink_message_exit_container(m);
+                }
+        } else
+                link->bridge_port = false;
```

Comment:
And the case when `IFLA_MASTER` is present, but `IFLA_BRPORT_STATE` is missing? Shouldn't `bridge_port` get set to false in that case?